campaigns/mobile/mobile-snippet-widget.html

```diff
@@ -0,0 +1,405 @@
+             }
+             else {
+                 button.disabled = true;
+                 button.classList.add('insensitive');
+             }
+         });
+
+        var evt = document.createEvent("HTMLEvents");
+        evt.initEvent("input", false, true);
+        input.dispatchEvent(evt);
+
+         // Submitting the form
+         var form = document.getElementById('send-to-device-form-{{ snippet_id }}');
+         button.addEventListener('click', function(event) {
+             event.preventDefault();
+             if (input_type === 'email') {
+                 sendMetric('conversion-mobile-activation-email', 
+                  function() { 
+                      sendEmailToBasket();
+                  });
+             } else if (input_type === 'phone') {
+                 sendMetric('conversion-mobile-activation-sms', 
+                  function() { 
+                      sendSMSToBasket();
+                  });
+             }
+         });
+
+         // Logo change.
+         {% if replacement_logo %}
+           var newLogo = '{{ replacement_logo }}';
+           var brandLogo = document.getElementById('brandLogo');
+           if (brandLogo.src !== undefined) {
+               brandLogo.src = newLogo;
+           } else {
+               brandLogo.style.backgroundImage = 'url(' + newLogo + ')';
+           }
+           brandLogo.className += ' switch-{{ snippet_id }}';
+         {% endif %}
+
+         // Check if it is a valid email or phone number or neither
+         function checkInput (inp) {
+              var email_re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+              var check_email = email_re.test(inp);
+              // Light check for a valid US phone number
+              var check_phone = inp.length>=10 && !isNaN(inp);
+
+              if (check_email) {
+                  return 'email';
+              } else if (check_phone && '{{ include_sms }}' === 'True') {
+                  return 'phone';
+              } else {
+                  return false;
+              }
+         }
+
+         function sendSMSToBasket () {
+              var url = 'https://basket.mozilla.org/news/subscribe_sms/';
+              var request = new XMLHttpRequest();
+
+              var params = 'mobile_number=' + input.value + '&msg_name={{ sms_id }}';
+
+              request.onreadystatechange = function() {
+                  if (request.readyState == 4) {
+                    changeState(request.status);
+                  }
+              }
+              request.open('POST', url);
+              request.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
+              request.send(params);
+         }
+
+         function sendEmailToBasket () {
+              //TODO: get language from data-locale
+              var lang = '{{ language }}';
```

Comment:
Thanks Giorgos. We are adding language as part of the template for ease of
testing other languages for the snippet admins. I'll have to come up with a
better way in the future, and will note this.

On Thursday, November 12, 2015, Giorgos Logiotatidis <
notifications@github.com> wrote:

> In campaigns/mobile/mobile-snippet-widget.html
> <https://github.com/mozilla/snippets/pull/60#discussion_r44665969>:
>
> > +
> > +              var params = 'mobile_number=' + input.value + '&msg_name={{ sms_id }}';
> > +
> > +              request.onreadystatechange = function() {
> > +                  if (request.readyState == 4) {
> > +                    changeState(request.status);
> > +                  }
> > +              }
> > +              request.open('POST', url);
> > +              request.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
> > +              request.send(params);
> > +         }
> > +
> > +         function sendEmailToBasket () {
> > +              //TODO: get language from data-locale
> > +              var lang = '{{ language }}';
>
> document.getElementById('snippet_set').getAttribute('data-locale');
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/snippets/pull/60/files#r44665969>.
>


-- 
Cory Price
/ckprice